rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,159 @@
+deleted. This method relies on identifying the exact position within the data file, thereby providing a more precise and
+direct approach to managing row-level deletions. Position deletes can be further categorized into two representations: 
+simple table representations and compressed bitmap representations.
+
+**_Simple table representations_** involve a straightforward structure typically stored in Parquet format, containing 
+columns like `file_path` and `pos`. The `file_path` indicates the path of the data file, while `pos` specifies the 
+ordinal of the record to be deleted. These files can optionally include the row data for equality deletes, aiding in 
+identifying the exact change which is particularly useful in CDC applications.
+
+A **_compressed bitmap representations_**, leverage a compressed bitmap to represent deletions. This method allows for a
+more compact and optimized storage of delete vectors, which in turn saves memory and network bandwidth. The 
+_RoaringBitmap_ is a popular choice for this representation. While the bitmaps are typically stored in separate files, 
+in cases where the bitmaps are small because the number of deleted records is low, they can be embedded within the 
+commit logs to optimize the number of network round trips required to load all the files.
+
+#### Delta Lake and Iceberg
+Currently, Delta Lake supports RoaringBitmaps for deletion vectors, while Iceberg supports simple table representations.
+The java libraries of Delta Lake and Iceberg provide APIs to read and write between these representations. Delta Lake 
+provides support for deletion vectors for `DELETE` operations since versions 2.4, while `UPDATE` and `MERGE` commands 
+default to the copy-on-write mode.
+
+In Delta Lake, the implementation of deletion vectors involves `actions` in the commit logs, namely `AddFile` and 
+`RemoveFile`. When a DML command is executed, it identifies the rows to be modified and generates a deletion vector, 
+typically in a compressed bitmap format like RoaringBitmap. The generated deletion vector is stored directly 
+in the commit logs `json` files (when the number of deletions is minimal). When stored in separate files, the files 
+paths are encoded and referenced in the json files. With each DML operation, the commit log records the removal of the 
+original data file and its addition with updated details including the newly added deletion vector. 
+
+Example:
+```
+Type       Path           DV Path
+RemoveFile file_a.parquet NULL
+AddFile    file_a.parquet file_a_dv_1.bin
+```
+
+If a file with an existing deletion vector is updated again, a new deletion vector is created, merging the previous and 
+new deletions. The commit log records the removal of the old deletion vector information and the addition of the new 
+one.
+
+Example:
+```
+Type       Path           DV Path
+RemoveFile file_a.parquet file_a_dv_1.bin
+AddFile    file_a.parquet file_a_dv_2.bin
```

Comment:
Just to make sure I understand the description above correctly, `file_a_dv_2.bin` also contains the deletions in `file_a_dv_1.bin`?

Reply:
Yes, this is as per the spec. Here is the corresponding test case in XTable: [L144](https://github.com/apache/incubator-xtable/blob/main/xtable-core/src/test/java/org/apache/xtable/delta/ITDeltaDeleteVectorConvert.java#L144)